rosette/src/Import.cc

```diff
@@ -413,28 +452,45 @@ pOb createRosetteObject(ObjectCodePB::Object * ob) {
     return retval;
 }
 
-
+// This routine creates a Rosette Code object along with its respective CodeVec and LitVec
 Code * createRosetteCode(ObjectCodePB::CodeBlock * cb) {
-fprintf(stderr, "%s\n", __PRETTY_FUNCTION__);
 
-    // Create the CodeBuf and CodeVec objects, then copy in the opcodes. 
-    int codesize = cb->codevec().opcodes().size();   // In bytes
-    CodeBuf * cbuf = CodeBuf::create();
-    cbuf->growCodevec( codesize / sizeof(Instr) );  // Set the size in 16 bit words
-    char * code_binary = (char *)cbuf->codevec->absolutize(0);  // Pointer to the buffer to hold the code
-    memcpy(code_binary, cb->codevec().opcodes().data(), codesize);
+    if (VerboseFlag) {
+        fprintf(stderr, "%s:\nCode=\n%s\n", __PRETTY_FUNCTION__, cb->DebugString().c_str());
+    }
+
+    // Create the CodeBuf object, then copy in the opcodes. 
+    int codewords = cb->codevec().opcodes().size();   // In words
+    CodeBuf * cbuf = CodeBuf::create(); // Create an empty CodeBuf
+
+    // Import the object code
+    ObjectCodePB::CodeVec * cv = cb->mutable_codevec();
+    for(int i=0; i<codewords; i++) {
+        Instr ins;
+        ins.word = cv->opcodes(i);
+        cbuf->deposit( ins );
+    }
 
     // Create the LitVec object and populate it from the protobuf objects
-    Tuple * litvec = Tuple::create(cb->litvec().ob().size(), INVALID);
-    for (int i=0; i<cb->litvec().ob().size(); i++) {
-        ObjectCodePB::Object * pbob = cb->mutable_litvec()->mutable_ob(i);
+    Tuple * litvec = Tuple::create(cb->litvec().ob_size(), INVALID);
+    ObjectCodePB::LitVec * pblv = cb->mutable_litvec();
+    for (int i=0; i<pblv->ob_size(); i++) {
+        ObjectCodePB::Object * pbob = pblv->mutable_ob(i);
         litvec->setNth(i, createRosetteObject(pbob));
     }
 
+    // Create the Code object with the CodeBuf and LitVec
     Code * code = Code::create(cbuf, litvec);
+
+    // if (VerboseFlag) {
```

Comment:
cruft

Reply:
I know.  I want to add this back later when I add code execution. It's commented out for now because it causes crashes due to the lack of globals being created during execution.